src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs

```diff
@@ -944,5 +944,57 @@ class Inner : IWhatever {
             await TestMoveTypeToNewFileAsync(
                 code, codeAfterMove, expectedDocumentName, destinationDocumentText);
         }
+
+        [WorkItem(17930, "https://github.com/dotnet/roslyn/issues/17930")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task MoveTypeWithDirectives1()
+        {
+            var code =
+@"using System;
+
+namespace N
+{
+    class Program
+    {
+        static void Main()
+        {
+        }
+    }
+}
+
+#if true
+public class [||]Inner
+{
+
+}
+#endif";
+var codeAfterMove =
+@"using System;
+
+namespace N
+{
+    class Program
+    {
+        static void Main()
+        {
+        }
+    }
+}
+
+#if true
```

Comment:
Yes.  But at least it isn't broken code :)

Actually making it so we can remove these directives is non-trivial as we use the SyntaxEditor API and it has no facility for this.